vcsi/vcsi.py

```diff
@@ -1325,8 +1326,30 @@ def process_file(path, args):
         args.metadata_horizontal_margin = args.metadata_margin
         args.metadata_vertical_margin = args.metadata_margin
 
+    if not args.grid:
+        args.grid = DEFAULT_GRID_SIZE
+
     args.num_selected = args.grid.x * args.grid.y
 
+    # manual frame selection
+    if args.manual_timestamps is not None:
+        mframes_size = len(args.manual_timestamps)
+        grid_size = args.grid.x * args.grid.y
+
+        args.num_selected = mframes_size
+        args.num_samples = mframes_size
+
+        if not mframes_size == grid_size:
+            # specified number of columns
+            y = math.ceil(mframes_size / args.grid.x)
+            args.grid = Grid(args.grid.x, y)
+
+    # interval frame selection
+
+    if args.num_selected < 1:
```

Comment:
Great, go ahead and make whatever changes you need.

I will put in some tests for the edge cases.